protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/plugin/BinaryPluginResolver.java

```diff
@@ -97,40 +102,56 @@ private ResolvedProtocPlugin resolveMavenPlugin(
 
     plugin = pluginBuilder.build();
 
+    log.debug("Resolving Maven protoc plugin {}", plugin);
+
     // Only one dependency should ever be returned here.
     var path = dependencyResolver.resolveOne(plugin, DependencyResolutionDepth.DIRECT)
         .iterator()
         .next();
 
     makeExecutable(path);
-    return createResolvedProtocPlugin(plugin, path);
+    var resolvedPlugin = createResolvedProtocPlugin(plugin, path);
+    return Optional.of(resolvedPlugin);
   }
 
-  private ResolvedProtocPlugin resolvePathPlugin(
+  private Optional<ResolvedProtocPlugin> resolvePathPlugin(
       PathProtocPlugin plugin
   ) throws ResolutionException {
-    var path = systemPathResolver.resolve(plugin.getName())
-        .orElseThrow(() -> new ResolutionException(
-            "No executable '"
-                + plugin.getName()
-                + "' was found on the system path"
-        ));
-
-    return createResolvedProtocPlugin(plugin, path);
+
+    log.debug("Resolving Path protoc plugin {}", plugin);
+
+    var maybePath = systemPathResolver.resolve(plugin.getName());
+
+    if (maybePath.isEmpty() && plugin.isOptional()) {
+      return Optional.empty();
+    }
+
+    var path = maybePath.orElseThrow(() -> new ResolutionException(
+        "No plugin named " + plugin.getName() + " was found on the system path"
+    ));
+
+    return Optional.of(createResolvedProtocPlugin(plugin, path));
   }
 
-  private ResolvedProtocPlugin resolveUrlPlugin(
+  private Optional<ResolvedProtocPlugin> resolveUrlPlugin(
       UrlProtocPlugin plugin
   ) throws ResolutionException {
-    var path = urlResourceFetcher
-        .fetchFileFromUrl(plugin.getUrl(), ".exe")
-        .orElseThrow(() -> new ResolutionException(
-            "Resource at " + plugin.getUrl() + " does not exist"
-        ));
+
+    log.debug("Resolving URL protoc plugin {}", plugin);
+
+    var maybePath = urlResourceFetcher.fetchFileFromUrl(plugin.getUrl(), ".exe");
+
+    if (maybePath.isEmpty() && plugin.isOptional()) {
```

Comment:
might refactor this and the above into one function in the future... not changing in this PR.